packages/labs/task/src/test/task_test.ts

```diff
@@ -942,4 +942,19 @@ suite('Task', () => {
     assert.equal(el.task.value, secondValue);
     assert.notEqual(firstValue, secondValue);
   });
+
+  test('type-only render return type test', () => {
+    class TestElement extends ReactiveElement {
+      task = new Task(this, () => 'abc');
+    }
+    customElements.define(generateElementName(), TestElement);
+    const el = new TestElement();
+
+    const acceptNumberOrUndefined = (x?: number) => x;
+
+    acceptNumberOrUndefined(el.task.render({initial: () => 123}));
+    acceptNumberOrUndefined(el.task.render({complete: () => 123}));
+    acceptNumberOrUndefined(el.task.render({pending: () => 123}));
+    acceptNumberOrUndefined(el.task.render({error: () => 123}));
```

Comment:
let's add a test for when two callbacks are passed (to ensure it works when there are multiple), and for when all four are passed

also, when all four are passed, I think we can infer the return type as `number`

https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAJwKZgCauQHgCqKoAeU6GAzogN4A2qwUA-ABQCUAXIuANZhwDuYADQoYAcwAWTNpx59BAXwB8zAG6c8ragChEiGMETMAsgEMoEgHTJTmOAFs2Wqrr0pUUEMiSrr4qYyWbIimlGYAngBGqABKHl5geOEADqj4ANoA5Mj+UJkAukoA3K4Krmie3oi+dAyBwaGIEdFxlYkpaXhZtXmFJWXaUB1NplGx8d5JqTgAYkqIALyIM4QkZJTMllumyGLknLbhrAtKMGDA2IgxiIxXiLKY9GeoGCXaEAjkUNWL7o-IzFo9CgnGCJ0QAEYAAwiHKSEFGLTggBMUIUrBKAHpMW5EAA9RjabTEZJwZDfKgKIA